### PR TITLE
Multiple Policy Updates - MDFC, TLS, CMK

### PR DESCRIPTION
**Pull request by Springstone**
## Overview/Summary

Providing updates aligned with internal user stories for deprecated policies in existing initiatives and adding new features for the Microsoft Defender initiative (API and CSPM) capability. Includes portal experience UI updates, other RIs need separate updates.

Addresses:
- [AB#25987](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/25987) and GH Issue #1159 
- [AB#26064](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/26064)
- [AB#26059](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/26059)
- [AB#25197](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/25197)

## This PR fixes/adds/changes/removes

- Removed deprecated policy [[Deprecated]: Latest TLS version should be used in your API App (azadvertizer.net)](https://www.azadvertizer.net/azpolicyadvertizer/8cb6aa8b-9e41-4f4e-aa25-089a7ac2581e.html) from initiative [Deny or Deploy and append TLS requirements and SSL enforcement on resources without Encryption in transit (azadvertizer.net)](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-EncryptTransit.html) as recommended policy is already included in the initiative.
- Updated initiative [Deny or Audit resources without Encryption with a customer-managed key (CMK) (azadvertizer.net)](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Enforce-Encryption-CMK.html) deprecated policy [[Deprecated]: SQL servers should use customer-managed keys to encrypt data at rest](https://www.azadvertizer.net/azpolicyadvertizer/0d134df8-db83-46fb-ad72-fe0c9428c8dd.html) to new policy [Azure Policy definition SQL servers should use customer-managed keys to encrypt data at rest](https://www.azadvertizer.net/azpolicyadvertizer/0a370ff3-6cab-4e85-8995-295fd854c5b8.html)
- Updated intiative and assignment [Deploy Microsoft Defender for Cloud configuration](https://www.azadvertizer.net/azpolicyinitiativesadvertizer/Deploy-MDFC-Config.html) to include the new policies:
  - [[Preview]: Configure Microsoft Defender for APIs should be enabled](https://www.azadvertizer.net/azpolicyadvertizer/e54d2be9-5f2e-4d65-98e4-4f0e670b23d6.html)
  - [Configure Microsoft Defender CSPM to be enabled](https://www.azadvertizer.net/azpolicyadvertizer/689f7782-ef2c-4270-a6d0-7664869076bd.html)
  - [Configure machines to receive a vulnerability assessment provider](https://www.azadvertizer.net/azpolicyadvertizer/13ce0167-8ca6-4048-8e6b-f996402e3c1b.html)

### Breaking Changes

1. These changes should not impact existing deployments.

## Testing Evidence

Please provide any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

### Testing URLs

The below URLs can be updated where the placeholders are, look for `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}` & `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}`, to allow you to test your portal deployment experience.

> Please also replace the curly brackets on the placeholders `{}`

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fspringstone%2FEnterprise-Scale%2Fsachana-policyupdates%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fspringstone%2FEnterprise-Scale%2Fsachana-policyupdates%2FeslzArm%2Feslz-portal.json)

#### Azure US Gov (Fairfax)
[Deploy To Azure](https://portal.azure.us/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2F{YOUR GITHUB ORG/ACCOUNT HERE - Remove Curly Brackets Also}%2FEnterprise-Scale%2F{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2F{YOUR GITHUB ORG/ACCOUNT HERE - Remove Curly Brackets Also}%2FEnterprise-Scale%2F{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}%2FeslzArm%2Ffairfaxeslz-portal.json)

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [ ] Performed testing and provided evidence.
- [X] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [X] Updated relevant and associated documentation.
- [X] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-MDFC-Config.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "displayName": "Deploy Microsoft Defender for Cloud configuration",
     "description": "Deploy Microsoft Defender for Cloud configuration",
     "metadata": {
-      "version": "3.1.1",
+      "version": "4.1.1",
```
Comment (jtracey93) [inline src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-MDFC-Config.json:11]:
```suggestion
      "version": "4.0.0",
```

When you increase a major/minor, you always reset the others to 0 👍

File: src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "displayName": "Deny or Audit resources without Encryption with a customer-managed key (CMK)",
     "description": "Deny or Audit resources without Encryption with a customer-managed key (CMK)",
     "metadata": {
-      "version": "1.0.1",
+      "version": "2.0.1",
```
Comment (jtracey93) [inline src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.json:11]:
```suggestion
      "version": "2.0.0",
```

as above

File: src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.AzureUSGovernment.json [hunk 3]
```diff
@@ -134,9 +134,10 @@
       },
       "SqlServerTDECMKEffect": {
         "type": "String",
-        "defaultValue": "AuditIfNotExists",
+        "defaultValue": "Audit",
```
Comment (jtracey93) [inline src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.AzureUSGovernment.json:137]: version not increased on line 11?

File: src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.AzureChinaCloud.json [hunk 4]
```diff
@@ -158,9 +158,10 @@
       },
       "SqlServerTDECMKEffect": {
         "type": "String",
-        "defaultValue": "AuditIfNotExists",
+        "defaultValue": "Audit",
```
Comment (jtracey93) [inline src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.AzureChinaCloud.json:161]: version not increased on line 11?

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Looks good, just some minor changes please @Springstone

File: src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.AzureUSGovernment.json (hunk 3 above)
Comment (Springstone) [reply src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.AzureUSGovernment.json:137]: My bad, missed this one, it is now fixed.

File: src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.AzureChinaCloud.json (hunk 4 above)
Comment (Springstone) [reply src/resources/Microsoft.Authorization/policySetDefinitions/Enforce-Encryption-CMK.AzureChinaCloud.json:161]: My bad, missed this one, it is now fixed.

**Comment by Springstone** [comment]
@jtracey93 I have provided the updates as requested.

**Comment by Springstone** [comment]
@jtracey93 Good to merge?

**Comment by Springstone** [comment]
@JefferyMitchell this is a merge into policy-refresh branch, not to main (live)

**Comment by Springstone** [comment]
@JefferyMitchell kindly expedite as this is impacting additional merges.

**Review by jtracey93 — APPROVED** [review]
LGTM will run tests

**Review by jtracey93 — APPROVED** [review]

**Comment by jtracey93** [comment]
Test deployed successfully, destroy failing as per known bug (cleaning up manually). Will merge

**Pull request merged**